vscode/src/chat/chat-view/ChatsController.ts

```diff
@@ -0,0 +1,479 @@
+    private editors: ChatController[] = []
+    private activeEditor: ChatController | undefined = undefined
+
+    // View controller for the support panel
+    public supportTreeViewProvider = new TreeViewProvider('support', featureFlagProvider)
+
+    // We keep track of the currently authenticated account and dispose open chats when it changes
+    private currentAuthAccount: undefined | { endpoint: string; primaryEmail: string; username: string }
+
+    protected disposables: vscode.Disposable[] = []
+
+    constructor(
+        private options: Options,
+        private chatClient: ChatClient,
+        private readonly enterpriseContext: EnterpriseContextFactory,
+        private readonly localEmbeddings: LocalEmbeddingsController | null,
+        private readonly contextRanking: ContextRankingController | null,
+        private readonly symf: SymfRunner | null,
+        private readonly guardrails: Guardrails
+    ) {
+        logDebug('ChatsController:constructor', 'init')
+        this.panel = this.createChatController()
+    }
+
+    public async syncAuthStatus(authStatus: AuthStatus): Promise<void> {
+        const hasLoggedOut = !authStatus.isLoggedIn
+        const hasSwitchedAccount =
+            this.currentAuthAccount && this.currentAuthAccount.endpoint !== authStatus.endpoint
+        if (hasLoggedOut || hasSwitchedAccount) {
+            this.disposeAllChats()
+        }
+
+        const endpoint = authStatus.endpoint ?? ''
+        this.currentAuthAccount = {
+            endpoint,
+            primaryEmail: authStatus.primaryEmail,
+            username: authStatus.username,
+        }
+
+        this.supportTreeViewProvider.syncAuthStatus(authStatus)
+
+        this.panel.syncAuthStatus()
+    }
+
+    public async restoreToPanel(panel: vscode.WebviewPanel, chatID: string): Promise<void> {
+        try {
+            await this.getOrCreateEditorChatController(chatID, panel.title, panel)
+        } catch (error) {
+            logDebug('ChatsController', 'restoreToPanel', { error })
+
+            // When failed, create a new panel with restored session and dispose the old panel
+            await this.getOrCreateEditorChatController(chatID, panel.title)
+            panel.dispose()
+        }
+    }
+
+    public registerViewsAndCommands() {
+        this.disposables.push(
+            vscode.window.registerTreeDataProvider(
+                'cody.support.tree.view',
+                this.supportTreeViewProvider
+            ),
+            vscode.window.registerWebviewViewProvider('cody.chat', this.panel, {
+                webviewOptions: { retainContextWhenHidden: true },
+            })
+        )
+
+        const debouncedRestoreToEditor = debounce(
+            async (chatID: string, chatQuestion?: string) => {
+                try {
+                    logDebug('ChatsController', 'debouncedRestorePanel')
+                    await this.getOrCreateEditorChatController(chatID, chatQuestion)
```

Comment:
```suggestion
                    return this.getOrCreateEditorChatController(chatID, chatQuestion)
```

I think we will need to return this for agent